### DISCUSSION TITLE
Is there anyway to delay a redirect after a successful form submission?

### DISCUSSION THREAD

**Discussion by scajal** [inertiajs/inertia#1655] [Help (React)]
I'm using the React adapter with Laravel and I have a modal where:

- I have a form which uses the `useForm` helper.
- The backend returns a redirection response.
- I want to close the modal before Inertia redirects the user to the new page.

The question is, can I delay the redirect until the modal is closed?
I'm toggling the modal's state in the `onSuccess` event of the form, but it seems to just redirect immediately.

**Comment by scajal** [comment]
The redirect is automatically Made by Inertia because of the X-Inertia-Location header.

**Comment by reinink** [comment]
Interesting question! Right now when an Inertia response comes back from the server, Inertia transitions to that new page immediately. There isn't any way to delay that from happening when navigating to a different page.

Closing a dialog is an interesting use-case. Not even sure how I'd go about handling this. Part of me thinks we could create the ability to register some type of callback that can be run before navigating away, which could include a delay in it.

TLDR; Not possible today, but an interesting idea for a future improvement to the project.

**Reply by scajal** [reply]
Maybe by passing a promise to the `onSuccess` callback Inertia can wait for it to resolve and redirect afterwards?

**Reply by reinink** [reply]
Yeah exactly, something like that 👍

**Reply by lucasdcrk** [reply]
+1 for this idea, this would also allow to remove redirection all together when having a form with 'Save' and 'Save and continue editing' options.
